include/modules/sway/layout.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#ifdef FILESYSTEM_EXPERIMENTAL
+#include <experimental/filesystem>
+#else
+#include <filesystem>
+#endif
+#include <fmt/format.h>
+#include <unordered_map>
+#include <tuple>
+#include <fstream>
+#include <regex>
+#include "ALabel.hpp"
+#include "bar.hpp"
+#include "client.hpp"
+#include "modules/sway/ipc/client.hpp"
+#include "util/json.hpp"
+#include "util/sleeper_thread.hpp"
+#include <pugixml.hpp>
+
+namespace waybar::modules::sway {
+
+#ifdef FILESYSTEM_EXPERIMENTAL
+namespace fs = std::experimental::filesystem;
+#else
+namespace fs = std::filesystem;
+#endif
+
+class Layout : public ALabel, public sigc::trackable {
+ public:
+  Layout(const std::string&, const Json::Value&);
+  ~Layout() = default;
+  auto update() -> void;
+
+ private:
+
+  using ShortNames = std::tuple<std::string, std::string>;
+  using MemoizedShortNames = std::unordered_map<
+    std::string,
+    std::tuple<std::string, std::string>
+  >;
+
+  static inline const fs::path xbk_file_ = "/usr/share/X11/xkb/rules/evdev.xml";
```

Comment:
Should this be `xkb_file_`?

Also, kind of hardcoded, isn't it? May this file be at some different path? Like under another prefix?

Reply:
This path is defined and standart by https://linux.die.net/man/7/xkeyboard-config . I fairly assume there will be an issue only with nixos. Are you using nixos, and if so, have you some suggestions for a workaround ?

Reply:
You can get the path to the `/usr/share/X11/xkb` dir with `pkg-config xkeyboard-config --variable xkb_base`.

Reply:
Also, is it intentional or typo `xkb_file_` -> `xbk_file_`?

Reply:
The type is not intended. I will fix it, thanks for the feedback.
I suppose it is possible for meson.build to generate a macro which contains the path `XKB_BASE`. I don't know how to use meson.build properly, I'll explore a bit, I'll keep you informed.